### PR TITLE
docs: add a java guide

### PR DESCRIPTION
**Pull request by nicks**
Hello @ellenkorbes, @maiamcc,

Please review the following commits I made in branch nicks/ch4188/java-guide:

b327d3090df30b35f0109b7ebb5e57e591e66da5 (2020-03-17 19:32:51 -0400)
docs: add a java guide

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: docs/example_java.md [hunk 1]
```diff
@@ -3,9 +3,314 @@ title: "Example: Java"
...
+}
+```
+
+To start this server on Kubernetes, we need 3 configs:
+
+1) A [Dockerfile](https://github.com/windmilleng/tilt-example-java/blob/master/0-base/Dockerfile) that builds the image
+
+2) A [Kubernetes deployment](https://github.com/windmilleng/tilt-example-java/blob/master/0-base/kubernetes.yaml) that runs the image
+
+3) And finally, a [Tiltfile](https://github.com/windmilleng/tilt-example-java/blob/master/0-base/Tiltfile) that ties them together
```
Comment (maiamcc) [inline docs/example_java.md]:
```suggestion
3) And finally, a [Tiltfile](https://github.com/windmilleng/tilt-example-java/blob/master/0-base/Tiltfile) that ties them together:
```

File: docs/example_java.md [hunk 2]
```diff
@@ -3,9 +3,314 @@ title: "Example: Java"
...
+
+3) Download all the Spring dependencies.
+
+4) Compile the Java jar from scratch.
+
+But the Java community has done a lot of work to make caching dependencies and
+incremental compiles fast.  How can we better use the tools how they're meant
+to be used?
+
+With `local_resource`, we can compile the executable Jar locally, and copy it
```
Comment (maiamcc) [inline docs/example_java.md]: some users will want to compile the executable in the container for uniformity reasons--so maybe worth noting something like "If you feel strongly about compiling the executable inside the container , e.g. to maintain uniformity across your team, you can still shave off a lot of time here with Dockerfile optimizations, see [some blog post on the internet about this]".

File: docs/example_java.md [hunk 3]
```diff
@@ -3,9 +3,314 @@ title: "Example: Java"
...
+| Local Compile | 13.4s |
+{:.benchmark-report}
+
+
+## Step 3: Let's Live Update It
+
+When we make a change to a file, we currently have to build an image, deploy new Kubernetes configs,
+and wait for Kubernetes to schedule the pod.
+
+With Tilt, we can skip all of these steps, live-updating the pod in place.
```
Comment (maiamcc) [inline docs/example_java.md]: link to live update docs

File: docs/example_java.md (hunk 1 above)
Comment (nicks) [reply docs/example_java.md]: done

File: docs/example_java.md (hunk 3 above)
Comment (nicks) [reply docs/example_java.md]: done

**Pull request merged**